src/snowflake/cli/app/printing.py

```diff
@@ -98,12 +99,21 @@ def is_structured_format(output_format):
 
 def print_structured(result: CommandResult):
     """Handles outputs like json, yml and other structured and parsable formats."""
+    printed_end_line = False
     if isinstance(result, MultipleResults):
         _stream_json(result)
+    elif isinstance(result, StreamResult):
+        # A StreamResult prints each value onto its own line
+        # instead of joining all the values into a JSON array
+        for r in result.result:
+            json.dump(r, sys.stdout, cls=CustomJSONEncoder)
+            print(flush=True)
+            printed_end_line = True
```

Comment:
Won't `_stream_json` work?

Reply:
No, it wraps it in a json array, while I want one object per line

Reply:
No, it wraps it in a json array, while I want one object per line

Reply:
piping into JQ does work with multiple objects:
<img width="1370" alt="image" src="https://github.com/user-attachments/assets/356096fc-c5ce-46c5-a331-da296663b850">

and it's possible to collect the output back into an array with `jq -s`:
<img width="1369" alt="image" src="https://github.com/user-attachments/assets/34974614-f79b-4fb8-830f-d5d445b1a48e">

Reply:
TBH I think `_stream_json` could be replaced with just
```python
json.dump(list(result.result), sys.stdout, cls=CustomJSONEncoder, indent=4)
```